src/OrchardCore.Modules/OrchardCore.MiniProfiler/Startup.cs

```diff
@@ -33,7 +33,16 @@ public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder ro
             var store = serviceProvider.GetService<IStore>();
 
             var factory = store.Configuration.ConnectionFactory;
-            store.Configuration.ConnectionFactory = new MiniProfilerConnectionFactory(factory);
+            var profilerFactory = new MiniProfilerConnectionFactory(factory);
```

Comment:
@sebastienros @nakamura2000 

The new feature is great, I wanted it to do this when I wrote it but couldn't.

Happy with this in `Startup`, rather than the Factory?

Re: YesSql updates  must remember to update the `CreateIndex` methods to use the new one.

Reply:
> Happy with this in Startup, rather than the Factory?

Sorry but I couldn’t understand your suggestion point.

MiniProfiler has to change because `SqlDialectFactory.SqlDialects` became not public.

If `CommandInterpreterFactory.For` has Type parameter method overload , the code will be more simple.

Reply:
@nakamura2000 `SqlDialectFactory` has this: `public static ISqlDialect For(Type dbConnectionType)`, what else do you need?

Reply:
@sebastienros 
I think `CommandInterpreterFactory` have to add `CommandInterpreterFactory.For(Type dbConnectionType)`

Reply:
Doing already, should be there soon

Reply:
@nakamura2000 @sebastienros has added the missing feature to YesSql - he just needs to merge it to master to get it on NuGet.

Then we should be able to finish this one